src/service.ts

```diff
@@ -268,20 +269,15 @@ export class Service {
     output.setData(result);
   }
 
-  static createGist(json: object): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        const jsonURI = JSON.parse(this.response).html_url;
-        resolve(jsonURI);
-      });
-      xhr.addEventListener("error", function() {
-        reject();
-      });
-      xhr.open("POST", "https://api.github.com/gists", true);
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
+  static async createGist(json: object): Promise<string> {
+    const url = "https://api.github.com/gists";
+    const response = await fetch(url, {
+      method: "POST",
+      body: JSON.stringify(json),
+      headers: new Headers({ "Content-type": "application/json; charset=utf-8" })
     });
+    const result = JSON.parse(await response.text()).html_url;
+    return JSON.stringify(result);
```

Comment:
See my comment at https://github.com/wasdk/WebAssemblyStudio/pull/125#discussion_r169781671

> return `JSON.parse(await response.text()).html_url;` see removed code at line 276. also function no longer returns string, but needs too.

Reply:
can we use return type ` Promise<any>` here?

Reply:
why? `result` shall be string, if TS does not know about -- inform it via defining type of JSON parsing or variable itself.